### PR TITLE
RAMA U80-A, wilba.tech WT60-A, WT65-A, WT80-A, IS31FL3736 driver

### PR DESCRIPTION
**Pull request by Wilba6582**
This is a slightly premature first cut of the RAMA U80-A implementation.

Doing a PR now so @jackhumbert and @yiancar can see the IS31FL3736 driver implementation and discuss ideas about merging with the IS32FL3733 driver.

### CODE REVIEW

**Comment by yiancar** [comment]
I think the standard method as was done in rgb_matrix with 3733 should be applied here.

I think @jackhumbert you want to use this driver in the future?

File: keyboards/rama/u80_a/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,
+	    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+	    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS,
+	    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+	    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,
+	    KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS),
+
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```
Comment (noroadsleft) [inline keyboards/rama/u80_a/keymaps/default/keymap.c]: This code block should be removed, as this function is deprecated.

File: keyboards/rama/u80_a/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Keyboard Maintainer: [Wilba6582](https://github.com/Wilba6582)  
+Hardware Supported: RAMA U80-A PCB  
+Hardware Availability: [RAMA WORKS Store](https://ramaworks.store/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rama/u80_a:default
+
+See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (noroadsleft) [inline keyboards/rama/u80_a/readme.md]: Please replace this line with the following:

`See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).`

**Comment by jackhumbert** [comment]
I'm working on an implementation of the 3737 driver that uses a 12 led configuration instead of the 8 or 16 ones that the 3736 and the 3733 use - the only thing that's changing in both is the header file where the positions (`A_12`, etc) are defined. I think it might make sense if we have a common ISSI file that defines the common functions used here, and separate header files for each driver.

I'm not sure how we'll want to manage the mono vs RGB stuff here - maybe separate ISSI files for each? Very open to ideas here :)

**Comment by yiancar** [comment]
The struct idea is also a good one. whatever floats your boats guys. They all result in the same thing:)

**Comment by Wilba6582** [comment]
@yiancar @jackhumbert I propose changing matrix locations like `A_1` to be `SW1_CS1` so they match the pin names of the ICs exactly, and the register names, and the primary legends on the example matrix in the datasheets.

**Comment by Wilba6582** [comment]
@drashna This is ready to merge now. Deferring the refactor until after #3913 is closed (and I talk some sense into @yiancar )

**Comment by Wilba6582** [comment]
FWIW, WT60-A, WT65-A and WT80-A are three other PCBs I designed using the same design as the RAMA U80-A so they can all share a common code base.

**Comment by jackhumbert** [comment]
Awesome - thanks :)

**Pull request merged**